src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynRuleSetGenerator.cs

```diff
@@ -29,57 +29,27 @@ public class RoslynRuleSetGenerator
     {
         private const string SONARANALYZER_PARTIAL_REPO_KEY = "sonaranalyzer-{0}";
         private const string ROSLYN_REPOSITORY_PREFIX = "roslyn.";
+        private const string activeRuleText = "Warning";
+        private const string inactiveRuleText = "None";
 
         private readonly IAnalysisPropertyProvider sonarProperties;
-        private readonly string inactiveRuleActionText = GetActionText(RuleAction.None);
+        private readonly bool deactivateAll;
 
-        private string activeRuleActionText = GetActionText(RuleAction.Warning);
-        private RuleAction activeRuleAction = RuleAction.Warning;
-
-        public RoslynRuleSetGenerator(IAnalysisPropertyProvider sonarProperties)
+        public RoslynRuleSetGenerator(IAnalysisPropertyProvider sonarProperties, bool deactivateAll)
```

Comment:
let's put bool deactivateAll = false on the constructor, seems most of the calls are calling it with that value, so then you don't need to modify everything.